src/Database/Dialect/MysqlDialectTrait.php

```diff
@@ -47,7 +47,7 @@ trait MysqlDialectTrait
     /**
      * The schema dialect class for this driver
      *
-     * @var \Cake\Database\Schema\MysqlSchema
+     * @var \Cake\Database\Schema\Dialect\MysqlDialect
```

Comment:
Should this be `Cake\Database\Dialect\MysqlSchemaDialect` ? Four namespaces feels like a lot.

Reply:
I was torn on that.  The database namespaces are a bit of a mess IMO.

We have `Database\Driver` but then `Database\Dialect` which is a driver trait.  If we move `MysqlSchemaDialect` into `Database\Dialect` would that be clear?

It probably could be as long as we use the full name `MysqlSchemaDialect` and not `MysqlDialect`.

Reply:
The `MysqlDialectTrait` should probably be `MysqlDialectDriverTrait` to match.

Reply:
> Maybe it shouldn't even be a trait.

Probably not. The drivers wereobne of the places we thought that traits could  help fix deep inheritance trees. It has created another set of problems though :smile:

Reply:
I think the methods/properties in the trait should just be in the driver.  They aren't shared by anything.